SE-XXX Typed throws.md

```diff
@@ -502,6 +500,83 @@ In first place we will specify different examples where the Swift compiler inter
 
 `// TODO: I think we should put that to the detailed solution part now`
 
+### Reducing `catch` clause verbosity
+
+`// TODO: This is the \catch\ proposal, to be determined whether it should be includded in the final draft or not`
+
+From the side of the `catch` clause there are becoming verbose due to the fact of being the place where `error` is being casted from its primitive `Error` to a more specific `error`.
+Having to write `catch let error as FooError` seems a bit inconsistent with the rest of how `catch` works, as `error` is inferred to be a constant of type `Error`. The following example makes it pretty clear.
+
+```swift
+do { ... }
+// Here we have to explicitly declare `error` as a constant.
+catch let error as FooError { ... }
+// Whereas here you have `error` for free.
+catch { ... }
+```
+
+This unbalanced behaviour comming from the same reserved word can induce confusion when writing down different specific and general catch clauses, having to declare error by themselves in one case and not in the other.
+
+For this reason, we propose to simplify specific `catch` clauses to avoid the learning curve from a `catch`-all paradigm to a more enxhaustive one.
+
+The grammar is composed by the following:
+
+```swift
+catch ErrorType { /* error is ErrorType */ }
+```
+
+This comes handy with `class` and `struct` error types, but shines its best with `enum` types.
+
+```swift
+enum SomeErrors: Error { case foo, bar, baz }
+
+func theErrorMaker() throws SomeErrors
+
+do { try theErrorMaker() }
+catch .foo { ... }
+catch .bar { ... }
+catch .baz { ... }
+```
+
+This behaviour and syntax in general resembles a lot with how `switch` cases work because:
+
+1. The type can be ignored, as it is inferred.
+2. They must be exhaustive.
+
+In scenarios where different types are involved, each one has the same treatment from the grammar side:
+
+```swift
+
+do {
+    try throwsClass() // throws MyClass
+    try throwsStruct() // throws MyStruct
+    try throwsEnum() // throws MyEnum { case one, two }
+} 
+catch MyClass { ... }
+catch MyStruct { ... }
+catch .one { ... }
+catch .two { ... } 
+```
+
+And where multiple enums are being caught, it would be ony needed to specify the type of those cases that were repeated in every enum.

```

Comment:
```suggestion
And where multiple enums are being caught, it would be only needed to specify the type of those cases that were repeated in every enum.
```